### DISCUSSION TITLE
Uno support

### DISCUSSION THREAD

**Discussion by navio-n** [ab4d/Ab4d.SharpEngine.Samples#10] [General]
Any plans to support the Uno Platform?
Tnx

**Comment by abenedik** [comment]
Added support for Uno Platform is high on the todo list and it will be implemented in the future. 

Currently, you can use Ab4d.SharpEngine if you create Uno app that works only for WinUI.
In this case, add a reference to `Ab4d.SharpEngine` and `Ab4d.SharpEngine.WinUI`. Then you can create the SharpEngineSceneView object in the xaml.cs file and add it to your view - you can use the code from: [SharpEngineSceneViewInCode.xaml.cs](https://github.com/ab4d/Ab4d.SharpEngine.Samples/blob/main/Ab4d.SharpEngine.Samples.WinUI/QuickStart/SharpEngineSceneViewInCode.xaml.cs)

I would like to provide support for sharing the texture that is rendered by the Ab4d.SharpEngine with the target platform (in WinUI `SwapChainPanel` is used for that; on Android `SurfaceView` could be used, ...). I currently do not have time to understand better how to create Uno controls that would use different native controls that could be used for sharing texture. If you know of any good sample for that, I would be glad if you can share that.

Also, I do not know if there is any cross-plaform WritableBitmap object in Uno Platform. This would be very useful because it can be used as a fallback presentation type on platforms that do not support sharing textures.

Note that browser support is not possible because Vulkan cannot work there.

**Comment by abenedik** [comment]
Version v3.0 added basic support for Uno Platform for all desktop and mobile devices. To see that in action, start the Uno Platform sample in the `Ab4d.SharpEngine.Samples.UnoPlatform` folder.

The current version supports only using `SKBitmap` where the rendered texture from SharpEngine is copied from the GPU memory to the main CPU memory to the `SKBitmap` (calling `InstallPixels` method). This is not the most efficient method.

If you know a way how to share the rendered texture with Uno Platform, then please let us know and we will be glad to provide support for that. See also: https://github.com/unoplatform/uno/issues/19170
